docs/frameworks/nestjs.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Nest.js
+description: Learn how to integrate Featurevisor in Nest.js applications
+ogImage: /img/og/docs-frameworks-nest.png
+---
+
+Set up Featurevisor SDK instance in a Nest.js application using a custom middleware, including TypeScript integration. {% .lead %}
+
+## Hello World application
+
+Before going into Featurevisor integration, let's create a simple Hello World [Nest.js](https://nestjs.com/) application.
+
+We start by installing the Nest CLI:
+
+```bash
+$ npm install -g @nestjs/cli
+```
+
+Then we create a new Nest.js project:
+
+```bash
+$ nest new featurevisor-nest-app
+$ cd featurevisor-nest-app
+```
+
+## Featurevisor integration
+
+We install the Featurevisor SDK first:
+
+```bash
+$ npm install --save @featurevisor/sdk
+```
+
+We can now create an instance of the SDK and use it in our application:
+
+```typescript
+// main.ts
+import { NestFactory } from '@nestjs/core';
+import { AppModule } from './app.module';
+import { createInstance, FeaturevisorInstance } from '@featurevisor/sdk';
+
+async function bootstrap() {
+  const app = await NestFactory.create(AppModule);
+
+  const DATAFILE_URL = 'https://cdn.yoursite.com/datafile.json';
+  const REFRESH_INTERVAL = 60 * 5; // every 5 minutes
+
+  const f: FeaturevisorInstance = createInstance({
```

Comment:
TypeScript will automatically pick up the type `FeaturevisorInstance` from the usage of `createInstance()` function, so we don't need to type it here.